watertap/unit_models/tests/unit_test_harness.py

```diff
@@ -0,0 +1,268 @@
+
+    @pytest.fixture(scope="class")
+    def frame_unit(self):
+        self.configure_class()
+        return self.unit_model_block
+
+    @pytest.mark.unit
+    def test_unit_statistics(self, frame_unit):
+        blk = frame_unit
+        stats = blk._test_objs.stateblock_statistics
+
+        if number_variables(blk) != stats["number_variables"]:
+            raise UnitValueError(
+                "The number of variables were {num}, but {num_test} was "
+                "expected ".format(
+                    num=number_variables(blk), num_test=stats["number_variables"]
+                )
+            )
+        if number_total_constraints(blk) != stats["number_total_constraints"]:
+            raise UnitValueError(
+                "The number of constraints were {num}, but {num_test} was "
+                "expected ".format(
+                    num=number_total_constraints(blk),
+                    num_test=stats["number_total_constraints"],
+                )
+            )
+        if number_unused_variables(blk) != stats["number_unused_variables"]:
+            raise UnitValueError(
+                "The number of unused variables were {num}, but {num_test} was "
+                "expected ".format(
+                    num=number_unused_variables(blk),
+                    num_test=stats["number_unused_variables"],
+                )
+            )
+
+    @pytest.mark.unit
+    def test_units_consistent(self, frame_unit):
+        assert_units_consistent(frame_unit)
+
+    @pytest.mark.unit
+    def test_dof(self, frame_unit):
+        if degrees_of_freedom(frame_unit) != 0:
+            raise UnitAttributeError(
+                "The unit has {dof} degrees of freedom when 0 is required."
+                "".format(dof=degrees_of_freedom(frame_unit))
+            )
+
+    @pytest.mark.component
+    def test_initialization(self, frame_unit):
+        blk = frame_unit
+
+        # initialize
+        try:
+            blk.initialize(solver=blk._test_objs.solver, optarg=blk._test_objs.optarg)
+        except InitializationError:
+            raise InitializationError(
+                "The unit has failed to initialize successfully. Please check the output logs for more information."
+            )
+
+    @pytest.mark.component
+    def test_unit_solutions(self, frame_unit):
+        # self.configure_class()
+        blk = frame_unit
+        solutions = blk._test_objs.unit_solutions
+
+        # solve unit
+        if blk._test_objs.solver is None:
+            opt = get_solver()
+        else:
+            opt = get_solver(
+                solver=blk._test_objs.solver, options=blk._test_objs.optarg
+            )
+        results = opt.solve(blk)
+
+        # check solve
+        assert len(list(iscale.badly_scaled_var_generator(blk, zero=1e-8))) == 0
+        assert_optimal_termination(results)
+
+        # check results
+        for key, val in solutions.items():
+            pyo_obj = eval(key, {"m": self.model})
```

Comment:
```suggestion
            pyo_obj = self.model.find_component(key)
```
Per my suggestion above.